Fixes Elevator teleport when player is mounted on horse.

Xericore opened the pull request:
Hi,
I want the possibility to use the elevator while mounted on a horse. The horse should teleport with the player and the player should remain mounted after the teleport. I found there is already some code in Elevator.java that looks like it should handle this use case, but the horse doesn't teleport with the player.

Adding vehicle.eject(); before vehicle.teleport(); fixes that issue. The horse teleports to the destination, but the player is dismounted.

Unfortunately simply calling vehicle.addPassenger() it doesn't work. I had to resort to a BukkitRunnable().runTaskLater() solution to make it work. This is not very elegant, but I couldn't find a better solution. If you know a better implementation, I'd gladly hear about it.

Note that this code only executes when "smooth-movement" is set to false. I didn't bother looking at the option when "smooth-movement" is true because the mechanic is broken right now. See: https://dev.enginehub.org/youtrack/issue/CRAFTBOOK-3464

Best regards

me4502 commented:
Thanks, would it be possible to extract that method to one of the util functions, such as LocationUtil, and also use it in the Teleporter mechanic?

Xericore commented:
I'll have a look when I find some time, but I think it's possible.

Xericore commented:
I've extracted the method to LocationUtil and implemented the feature in the Teleporter as well. I also fixed the Elevator when smooth-movement is true (https://dev.enginehub.org/youtrack/issue/CRAFTBOOK-3464).
Smooth-movement also works when the player is riding a horse. In that case only the player will be moved smoothly, the horse/vehicle will be teleported. I will create a new pull request for all the fixes and features.

me4502 commented:
It says there are 0 commits?

Xericore commented:
Sorry give me a few minutes please. I think I messed up this pull request because I wanted to have a single commit. I will create a new pull request shortly.

Xericore commented:
Done, please review https://github.com/EngineHub/CraftBook/pull/220 when you find some time. Thanks.
